src/plugins/vis_builder/public/application/utils/validations/vis_builder_state_validation.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import Ajv from 'ajv';
+import visBuilderStateSchema from '../schema.json';
+import { ValidationResult } from './types';
+
+const ajv = new Ajv();
+const validateState = ajv.compile(visBuilderStateSchema);
+
+export const validateVisBuilderState = (visBuilderState: any): ValidationResult => {
+  const isVisBuilderStateValid = validateState(visBuilderState);
+  const errorMsg = validateState.errors
+    ? validateState.errors[0].instancePath + ' ' + validateState.errors[0].message
```

Comment:
Good point. I just moved this over from where it was being used (saved object loader), and there we were only outputting the first error like this. The main reason for this change was so that all the validation functions follow the same `ValidationResult` interface. In future we can always pass the whole errors object too if we have a need for that by extending the interface, but to keep the code/PR simple I just pulled in the error parsing into the file.